docs/en/reference/types.rst

```diff
@@ -909,3 +909,55 @@ hook it into the database platform:
 This would allow using a money type in the ORM for example and
 have Doctrine automatically convert it back and forth to the
 database.
+
+It is also possible to register type instances directly, in case you
+need to pass parameters to your instance::
+
+    <?php
+    namespace My\Project\Types;
+
+    use Doctrine\DBAL\Types\Type;
+    use Doctrine\DBAL\Platforms\AbstractPlatform;
+
+    final class EncryptedType extends StringType
+    {
+        public function __construct(
+            private string $cipher,
+            private string $secretName,
+            private SecretStore $secretStore
+        ) {
+        }
+
+        public function convertToPHPValue($value, AbstractPlatform $platform): string
+        {
+            return openssl_decrypt(
+                $value,
+                $this->cipher,
+                $this->secretStore->fetch($this->secretName)
+            );
+        }
+
+        public function convertToDatabaseValue($value, AbstractPlatform $platform): string
+        {
+            return openssl_encrypt(
+                $value,
+                $this->cipher,
+                $this->secretStore->fetch($this->secretName)
+            );
+        }
+
+        public function getName(): string
+        {
+            return 'encrypted';
+        }
+    }
+
+To do that, you can obtain the ``TypeRegistry`` singleton from ``Type``
+and register your type in it::
+
+    <?php
+    Type::getTypeRegistry()->addType('encrypted', new EncryptedType(
```

Comment:
This is a bit confusing. Why does the user need to specify the `'encrypted'` name of the custom type twice: once in the type itself and the other time during the registration?

It looks like we should deprecate `Type::getName()` and make sure no code relies on it.

Reply:
I don't know but it is indeed confusing, and apparently already challenged: https://github.com/doctrine/dbal/blob/34af8fbc885d4853f8a138cb8fe3dfa7b0626dcb/src/Types/Type.php#L107-L109

Reply:
After looking into it, it seems used for reverse engineering: https://github.com/doctrine/dbal/blob/34af8fbc885d4853f8a138cb8fe3dfa7b0626dcb/src/Platforms/AbstractPlatform.php#L440-L472